dev/com.ibm.ws.microprofile.config.1.1_fat/fat/src/com/ibm/ws/microprofile/config/fat/tests/BasicConfigTests.java

```diff
@@ -118,4 +140,61 @@ public static void tearDown() throws Exception {
         server.stopServer();
     }
 
+    /**
+     * Verify that the MP Config introspection is included in a server dump and includes expected config vars
+     */
+    @Test
+    @Mode(TestMode.FULL)
+    public void introspectionTest() throws Exception {
+        LocalFile file = server.dumpServer("introspectionTestDump");
+        Map<String, String> introspectionParts;
+        try (ZipFile zip = new ZipFile(file.getAbsolutePath())) {
+            Optional<? extends ZipEntry> entry = zip.stream().filter(e -> e.getName().matches(".*MicroProfileConfig\\.txt")).findFirst();
+            assertTrue("MP Config introspection missing from dump", entry.isPresent());
+            introspectionParts = readIntrospectionSections(zip.getInputStream(entry.get()));
+        }
+
+        for (Entry<String, String> entry : introspectionParts.entrySet()) {
+            assertThat(entry.getValue(), containsString("introspectorTest = env")); // from server.env
+            assertThat(entry.getValue(), containsString("introspectorTest = sysprops")); // from bootstrap.properties
+
+            if (entry.getKey().equals(CONVERTERS_APP_NAME)) {
+                assertThat(entry.getValue(), containsString("introspectorTest = appprops")); // from microprofile-config.properties in converters app
+            } else {
+                assertThat(entry.getValue(), not(containsString("introspectorTest = appprops"))); // not present in other apps
+            }
+        }
+    }
+
+    /**
+     * Read the introspection and split it into sections for each app
+     *
+     * @param in InputStream for introspection file
+     * @return map from app name to corresponding section of the introspection
+     */
+    private static Map<String, String> readIntrospectionSections(InputStream in) throws IOException {
+        Map<String, String> result = new HashMap<>();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
+        String line = reader.readLine();
+        StringBuilder currentBuilder = new StringBuilder();
+        String currentApp = null;
+        while (line != null) {
+            Matcher m = Pattern.compile("Config for (.+)").matcher(line);
```

Comment:
can the compile go outside the loop if this is being run in every full fat run?

Reply:
Yes, will do.